netket/experimental/driver/tdvp_common.py

```diff
@@ -62,7 +62,7 @@ def __init__(
         self,
         operator: AbstractOperator,
         variational_state: VariationalState,
-        solver: AbstractSolver,
+        solver: AbstractSolver = None,
         integrator: AbstractSolver = None,
         *,
         t0: float = 0.0,
```

Comment:
```suggestion
        solver: AbstractSolver = None,
        *,
        t0: float = 0.0,
        integrator: AbstractSolver = None,
```

as above. If the user specifies the solver positionally, we don't need to warn. It's only if integrator is specified as a keyword argument that we should

Reply:
and below you should raise a warning if `integrator is not None`

Reply:
Is the warning defined properly? (sorry, new to deprecations)

Reply:
the logic is good. I'd just change the text slightly (see below)